nagstamon: 3.14.0 -> 3.16.2

liberodark opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

FliegendeWurst commented on pkgs/tools/misc/nagstamon/default.nix:
```diff
@@ -1,21 +1,26 @@
+  # Use wrapQtAppsHook to wrap the app properly
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
   meta = with lib; {
```
Set platforms = linux. Doesn't build on Mac.

khaneliman replied:
Platforms should represent what upstream supports. If it doesn't build we should mark it as a badPlatforms. The error looks like it just might need tweaking to fix. 

'''bash

    >               ^^^^^^^^^^^^^^^^^
       > NameError: name 'bdist_rpm_options' is not defined. Did you mean: 'bdist_mac_options'?

liberodark replied:
Sorry but I don't quite understand what you expect from me.
Sorry the goal of this PR is to fix this application which has been broken for several versions of nixos.

liberodark replied:
Need to add just : platforms = platforms.linux; ?

Artturin commented on pkgs/tools/misc/nagstamon/default.nix:
```diff
@@ -1,25 +1,58 @@
+  # Correction : buildSystem -> nativeBuildInputs
+  nativeBuildInputs = with pythonPackages; [
+    setuptools
+    qt6.wrapQtAppsHook # Déplacé ici car c'est un outil de build
```
English please, the comment is unnecessary.

Artturin commented on pkgs/tools/misc/nagstamon/default.nix:
```diff
@@ -1,25 +1,58 @@
+    qt6.qtsvg
+  ];
+
+  # Correction : buildSystem -> nativeBuildInputs
```
Don't rename, those attr names are what the docs use. They probably want to do something under the hood with them in the future.

Keep the wrapQtAppsHook in nativeBuildInputs

SuperSandro2000 replied:
We are only using build-system when pyproject is set to true

Artturin commented on pkgs/top-level/all-packages.nix:
```diff
@@ -9417,6 +9417,7 @@ with pkgs;
 
   nagstamon = callPackage ../tools/misc/nagstamon {
     pythonPackages = python3Packages;
+    qt6 = kdePackages;
```
Why? You're not using anything specific to that set, use qt6Packages directly

GaetanLepage commented on pkgs/tools/misc/nagstamon/default.nix:
```diff
@@ -1,25 +1,55 @@
+  version = "3.16.2";
 
   src = fetchurl {
     url = "https://github.com/HenriWahl/Nagstamon/archive/refs/tags/v${version}.tar.gz";
-    sha256 = "sha256-9RxQ/rfvoyjSUsY4tmAkBdVQqZYi3X6PBzQYFIeenzA=";
+    sha256 = "sha256-QuXs0muY0CvEJFB7ehMhvcQG4MByp3ySEmr7qB1enEk=";
   };
```
Why not using `fetchFromGitHub` ?

GaetanLepage commented:
Ideally, the formatting changes should be isolated in a separate commit.

liberodark commented:
> Ideally, the formatting changes should be isolated in a separate commit.

Hi,
Can you make exception about this ?
Is easy to read all changes and have fix this broken app.
So I made all the requested changes.

Best Regards

GaetanLepage commented on pkgs/tools/misc/nagstamon/default.nix:
```diff
@@ -1,25 +1,57 @@
+  src = fetchFromGitHub {
+    owner = "HenriWahl";
+    repo = "Nagstamon";
+    rev = "v${version}";
```
```suggestion
    rev = "refs/tags/v${version}";
```
This prevents branch/tag mismatch.

GaetanLepage commented on pkgs/tools/misc/nagstamon/default.nix:
```diff
@@ -1,25 +1,57 @@
+{
+  lib,
+  fetchFromGitHub,
+  pythonPackages,
```
```suggestion
  python3Packages,
```

GaetanLepage commented on pkgs/tools/misc/nagstamon/default.nix:
```diff
@@ -1,25 +1,57 @@
-  dependencies = with pythonPackages; [ configparser pyqt6 psutil requests
-     beautifulsoup4 keyring requests-kerberos lxml dbus-python python-dateutil pysocks ];
+
+  dependencies = with pythonPackages; [
```
```suggestion
  dependencies = with python3Packages; [
```

GaetanLepage commented on pkgs/tools/misc/nagstamon/default.nix:
```diff
@@ -1,25 +1,57 @@
+
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
   build-system = with pythonPackages; [ setuptools ];
```
```suggestion
  build-system = with python3Packages; [ setuptools ];
```

GaetanLepage commented on pkgs/top-level/all-packages.nix:
```diff
@@ -9416,7 +9416,7 @@ with pkgs;
 
   nagstamon = callPackage ../tools/misc/nagstamon {
-    pythonPackages = python3Packages;
+    qt6 = qt6Packages;
```
Is there a reason for not directly using `qt6Packages` in the derivation inputs ?

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349714`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

GaetanLepage reviewed (approved):
LGTM !

liberodark commented:
Hi, 

Also if is possible to backport to 24.11 ?

Best Regards

liberodark commented on pkgs/by-name/na/nagstamon/package.nix:
```diff
@@ -0,0 +1,57 @@
+    qt6Packages.qtsvg
+  ];
+
+  nativeBuildInputs = [ qt6Packages.wrapQtAppsHook ];
```
Nop because is Qt app

FliegendeWurst replied:
I think the issue is that both the Python wrapper and the Qt wrapper will apply.

Therefore you should add
```
  dontWrapQtApps = true;

  preFixup = ''
    makeWrapperArgs+=("''${qtWrapperArgs[@]}")
  '';
```
to make sure it is only wrapped once.

GaetanLepage committed:
nagstamon: move to by-name

Co-authored-by: liberodark <liberodark@gmail.com>

GaetanLepage committed:
nagstamon: format

Co-authored-by: liberodark <liberodark@gmail.com>

GaetanLepage committed:
nagstamon: 3.14.0 -> 3.16.2

Co-authored-by: liberodark <liberodark@gmail.com>

Diff: https://github.com/HenriWahl/Nagstamon/compare/v3.14.0...v3.16.2

Changelog: https://github.com/HenriWahl/Nagstamon/releases/tag/v3.16.2

GaetanLepage commented:
I was not able to launch the application successfully on wayland:
```
❯ /nix/store/vd1d202vlfxhagxsx4hh6pfwlmafzd5i-nagstamon-3.16.2/bin/nagstamon.py 
[Errno 2] No such file or directory: '/home/gaetan/.nagstamon/nagstamon.pid'
qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: linuxfb, xcb, minimal, offscreen, minimalegl, vkkhrdisplay, vnc, eglfs.

fish: Job 1, '/nix/store/vd1d202vlfxhagxsx4hh…' terminated by signal SIGABRT (Abort)
```

liberodark commented:
Hi,

I dont have wayland sorry but i think this can be fixed.

Best Regards

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349714`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

GaetanLepage commented:
Fails on darwin with
```
Executing setuptoolsBuildPhase
Traceback (most recent call last):
  File "/private/tmp/nix-build-nagstamon-3.16.2.drv-0/source/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 162, in <module>
    bdist_rpm=bdist_rpm_options)
              ^^^^^^^^^^^^^^^^^
NameError: name 'bdist_rpm_options' is not defined. Did you mean: 'bdist_mac_options'?
```

liberodark commented:
Yes but have disable darwin plateform.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349714`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

GaetanLepage commented:
@Artturin could you please have another look at this PR ? It is good to go as far as I am concerned.

Artturin commented on pkgs/by-name/na/nagstamon/package.nix:
```diff
@@ -0,0 +1,67 @@
+  };
+
+  build-system = with python3Packages; [ setuptools ];
+
+  buildInputs = [
+    qt6Packages.qtmultimedia
+    qt6Packages.qtbase
+    qt6Packages.qtsvg
+    qt6Packages.qtwayland
+  ];
```
```suggestion
  nativeBuildInputs = [ qt6Packages.wrapQtAppsHook ];

  buildInputs = [
    qt6Packages.qtmultimedia
    qt6Packages.qtsvg
  ];
```

qtbase and qtwayland are propagated by the wrapper

GaetanLepage replied:
Thanks for the tip. Fixed.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349714`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

GaetanLepage commented:
cc @Artturin for final review
